lib/fauxpaas/task_file.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require "fauxpaas/tasks/shell"
+require "yaml"
+
+module Fauxpaas
+
+  # A list of tasks encoded in a file. This is typically used to
+  # translate e.g. finish_build.yml into a list of tasks.
+  class TaskFile
```

Comment:
I would be very tempted to make this a `Tasks` class.

Rather than use an initializer for reading and then `#tasks` for converting, it would have a `@tasks` array and a `.from_yaml` factory method to read/convert. It would be a typed collection, using `include Enumerable` and implementing `#each` to yield the tasks sequentially. The initializer would take a collection of concrete tasks and simply assign it, rather than attempting to convert them.

That does raise the question of the appropriate file/module layout, but I am happy with the answer. That is, `tasks.rb` and `task.rb` would be siblings (`Task` not being in the `Tasks` module), and then the concrete tasks would be things like `task/foo.rb` for `Task::Foo`. I think this is more conventional than the plural module name, and it is attractive to me to have the container and base class at the top level. `Fauxpaas::Tasks` and `Fauxpaas::Task` -- there just aren't that many names at the top level to justify pushing it down to me.